src/main/java/pl/karolpat/serviceImpl/ParkingMeterServiceImpl.java

```diff
@@ -0,0 +1,141 @@
+package pl.karolpat.serviceImpl;
+
+import java.sql.Timestamp;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import org.springframework.stereotype.Service;
+
+import pl.karolpat.entity.ParkingMeter;
+import pl.karolpat.entity.User;
+import pl.karolpat.repository.ParkingMeterRepo;
+import pl.karolpat.service.ParkingMeterService;
+
+/**
+ * @author karolpat
+ *
+ */
+@Service
+public class ParkingMeterServiceImpl implements ParkingMeterService {
+
+	private ParkingMeterRepo parkingMeterRepo;
+
+	static private String HOURS = "Hours spent";
+	static private String PLN_CURRENCY = "PLN";
+
+	public ParkingMeterServiceImpl(ParkingMeterRepo parkingMeterRepo) {
+		this.parkingMeterRepo = parkingMeterRepo;
+	}
+
+	@Override
+	public ParkingMeter save(ParkingMeter PM) {
+		parkingMeterRepo.save(PM);
+		return PM;
+	}
+
+	@Override
+	public ParkingMeter findUserParkingMeter(long id) {
+		return parkingMeterRepo.getFirstByUserIdOrderByStart(id);
+	}
+
+	@Override
+	public List<ParkingMeter> findAll() {
+		return parkingMeterRepo.findAll();
+	}
+
+	@Override
+	public Map<String, Double> checkCost(User user) {
+
+		int hours = getCurrentHours(user.getId()); // Number of hours that user spent at the parking.
+
+		Map<String, Double> map = new HashMap<>();
+		map.put(HOURS, (double) hours); // hours as a value to HOURS String key.
+
+		if (user.isVip() == true) { // Depending on User's vip status total cost will be different
+			map.put(PLN_CURRENCY, round(getCostIfVip(hours), 2));
+			return map;
+		} else {
+			map.put(PLN_CURRENCY, round(getCostUnlessVip(hours), 2));
+			return map;
+		}
+
+	}
+
+	@Override
+	public int getCurrentHours(long id) {
+
+		ParkingMeter current = parkingMeterRepo.getFirstByUserIdOrderByStart(id);// First ParkingMeter found by user id ordered by start date.
+
+		// Converts start date to minutes.
+		long start = TimeUnit.MINUTES.convert(current.getStart().getTime(), TimeUnit.MILLISECONDS);
+
+		// Converts end date (that is actually current time) to minutes.
+		long now = TimeUnit.MINUTES.convert(new Timestamp(System.currentTimeMillis()).getTime(), TimeUnit.MILLISECONDS);
+
+		int hours = (int) (now - start) / 60; // Converts difference of start and end time (minutes) to hours.
```

Comment:
czemu nie użyć Duration?

Reply:
Przyznam, że nie znałem tego, jednak jak zobaczyłem, to robi dokładnie to co napisałem. Dzięki